python/src/aac/io/writer.py

```diff
@@ -0,0 +1,60 @@
+"""AaC Module for easy file writing."""
+
+import logging
+from os import path, makedirs
+
+from aac.lang.definitions.definition import Definition
+from aac.io.constants import YAML_DOCUMENT_SEPARATOR
+from aac.io.paths import sanitize_filesystem_path
+
+
+def write_file(uri: str, content: str, overwrite: bool = False) -> None:
+    """
+    Write string content to a file.
+
+    Args:
+        uri (str): the file's full uri
+        content (str): contents of the file to write
+        overwrite (bool): True to overwrite an existing file or false to not.
+    """
+    sanitized_uri = sanitize_filesystem_path(uri)
+    does_file_exist = path.exists(sanitized_uri)
+    file_parent_dir = path.dirname(sanitized_uri)
+
+    if not path.exists(file_parent_dir):
+        makedirs(file_parent_dir, exist_ok=True)
+
+    if not overwrite and does_file_exist:
+        logging.info(f"{sanitized_uri} already exists, skipping write.")
+        return
+
+    try:
+        with open(sanitized_uri, "w") as file:
+            file.writelines(content)
+    except IOError as error:
+        logging.error(f"Failed to write file {sanitized_uri} due to error: {error}")
+
+
+def write_definitions_to_file(definitions: list[Definition], file_uri: str, is_user_editable: bool = True) -> None:
+    """
+    Given a list of definitions, write them to file uri. Updates the source for definitions passed in.
+
+    Args:
+        definitions (list[Definition]): The definitions to write to file.
+        file_uri (str): The URI of the file to write the definitions to.
+        is_user_editable (bool): True if the AaC file can be edited by users.
+    """
+    def sort_definitions_by_lexeme_line(definition: Definition) -> int:
+        return -1 if not definition.lexemes else definition.lexemes[0].location.line
```

Comment:
When would a definition not have lexemes? Is this mainly to account for definitions created for tests?

Reply:
While the normal flow usually ensures that lexemes are present, there is no step or function that ensures it so we can't assumes that lexemes will be present.